src/ucs/sys/sock.c

```diff
@@ -366,14 +370,33 @@ ucs_status_t ucs_socket_server_init(const struct sockaddr *saddr, socklen_t sock
         goto err_close_socket;
     }
 
-    status = ucs_sockaddr_get_port(saddr, &port);
-    if (status != UCS_OK) {
-        goto err_close_socket;
+    if ((ports_range_start != 0) && (ports_range_start == ports_range_end)) {
```

Comment:
can't we just check range end?
in any case need to check that range is valid (e.g start<=end<64k) and return error if not

Reply:
check what only in the range_end?
i need to tell here if it's a single value (not range) that isn't zero.

the check for range is valid is done in tcp_iface when doing the intersect with ip_local_port_range.
should this function check the same?

Reply:
if range_end is 0 than it's single value, no need to check start

Reply:
ok i got it
so we can write it as 
retry = (ports_range_start == 0) || (ports_range_start < ports_range_end)